File: 프로그래머스/unrated/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.java
import java.util.*;

class Solution {

    public int solution(int[] bandage, int health, int[][] attacks) {

        int curr = health;
        int attackIdx = 0;
        int healTime = 0;

        for (int i = 1; i <= attacks[attacks.length - 1][0]; i++) {
            // 공격받지 않은 경우
            if (i != attacks[attackIdx][0]) {
                curr += bandage[1];
                healTime += 1;
                if (healTime == bandage[0]) {
                    curr += bandage[2];
                    healTime = 0;
                }
                if (curr > health) {
                    curr = health;
                }
            } else { // 공격받은 경우
                curr -= attacks[attackIdx][1];
                healTime = 0;
                attackIdx += 1;
                if (curr <= 0) {
                    curr = -1;
                    break;
                }
            }
        }

        return curr;

    }
}

File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.py
def solution(bandage, health, attacks):
    
    last_time = attacks[-1][0]
    idx, cnt = 0, 0    
    curr_time = 0
    curr_health = health
    while curr_time <= last_time:
        if curr_time == attacks[idx][0]:
            cnt = 0
            curr_health -= attacks[idx][1]
            if curr_health <= 0:
                curr_health = -1
                break
            curr_time += 1
            idx += 1
            continue
        
        cnt += 1
        curr_health += bandage[1]                    
        if cnt == bandage[0]:
            cnt = 0
            curr_health += bandage[2]            
        if curr_health > health:
            curr_health = health
                
        curr_time += 1
        
    return curr_health